### ISSUE TITLE
Parts of published sims are minified that shouldn't be

### ISSUE THREAD

**Issue by jbphet** [phetsims/chipper#979]
In https://github.com/phetsims/website/issues/1123 it was reported that the translation utility isn't working for Natural Selection.  I started investigating and found that the problem was occurring during extraction of the strings information from the live HTML file.  When I examined the live HTML file, I found that the sections containing the license info, dependencies, strings, and possibly other portions were minified, which was not true in the past.  The translation utility has trouble finding the strings data structure in this situation.

We discussed this in today's developer meeting and decided that we should go back to not minifying these portions of the sim, since these sections are meant to be human readable.  @zepumph said that this change probably came about as part of the work to make sure that the "Not Supported" dialog would always work on IE.  He looked through the history in chipper and said that he found the commit where the problem was introduced, and it occurred on July 9th, so some other sims probably have this problem.  This means that a maintenance release will be required to correct the problem.

The SHA for chipper used in Natural Selection v 1.1.0 (the version that is live as of this writing and exhibits the problem) is 29b616a083a386ecb9744a5c55caccef1f2ab34d.

**Comment by jbphet** [comment]
I just checked two recently published sims - states-of-matter and molecules-and-light - and both also exhibit this problem, meaning that they can't be translated.

**Comment by zepumph** [comment]
Minification of the initialization script was added in (ish) https://github.com/phetsims/chipper/commit/e4162919be457f0e788533b08aefca666203c66c.

We should revert that and do an MR.

**Comment by zepumph** [comment]
Committed above. @jonathanolson I am planning to do an MR today.

**Comment by zepumph** [comment]
MR in progress, but it looks fixed in `view-source:https://phet-dev.colorado.edu/html/molecules-and-light/1.5.5-rc.2/phet/molecules-and-light_en_phet.html` and searching for `strings = `

Things are no longer minified.

**Comment by zepumph** [comment]
Alright here is how far I have gotten today. I wrote down every line of code I made for the MR, and took notes on any issues during:
```


Maintenance.list()
>>> Didn't run this: Maintenance.checkBranchStatus() because it was taking too long
Maintenance.reset()
Maintenance.createPatch( 'chipper', 'https://github.com/phetsims/chipper/issues/979', '')
Maintenance.addNeededPatchesAfter( 'chipper', 'e4162919be457f0e788533b08aefca666203c66c') #had to run a couple of times because of a bad working copy
Maintenance.checkoutBranch( 'atomic-interactions', '1.2' ) #didn't have to do this
Maintenance.addPatchSHA( 'chipper', 'd7aea3ba13a73243817dc1254640a8696e284c52' )
Maintenance.applyPatches()
did not get ph-scale or basics 1.4, or MAL 1.5
Maintenance.checkoutBranch( 'molecules-and-light', '1.5' )
git cherry-pick d7aea3ba13a73243817dc1254640a8696e284c52
---- fix merge conflict
git cherry-pick --continue
copy new sha made from cherry-pick (36aeba695faeccd1b2263fef3a023ddc2db94f64)
Maintenance.addPatchSHA( 'chipper', '36aeba695faeccd1b2263fef3a023ddc2db94f64' )
Maintenance.applyPatches()
# everything else was applied
Maintenance.updateDependencies()
#notified RC testing issues about the new RC
Maintenance.deployReleaseCandidates()

> error:

````>> Detected failure during deploy, reverting to master
Maintenance task failed:
Error: Failure with RC deploy for molecules-and-light to 1.5: Error: git push -u origin 1.5 in ../molecules-and-light failed with exit code 128 and stdout:

    at Function.deployReleaseCandidates (C:\Users\Michael\PhET\2git\perennial\js\common\Maintenance.js:786:17)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
Full Error details:
{}

Run
````
after I ran it again it worked:
Maintenance.deployReleaseCandidates()
```


Here is the output to Maintenance.list() on my working copy right now.

```
maintenance> Maintenance.list()
atomic-interactions 1.2 phet,phet-io (unreleased)
  pushedMessages: https://github.com/phetsims/chipper/issues/979
collision-lab 1.0 phet (unreleased)
  pushedMessages: https://github.com/phetsims/chipper/issues/979
energy-forms-and-changes 1.3 phet,phet-io (unreleased)
  pushedMessages: https://github.com/phetsims/chipper/issues/979
energy-forms-and-changes 1.4 phet,phet-io (unreleased)
  pushedMessages: https://github.com/phetsims/chipper/issues/979
energy-skate-park 1.0 phet,phet-io (unreleased)
  pushedMessages: https://github.com/phetsims/chipper/issues/979
energy-skate-park 1.1 phet,phet-io (unreleased)
  pushedMessages: https://github.com/phetsims/chipper/issues/979
molecules-and-light 1.5 phet
  deployed: 1.5.5-rc.2
  pushedMessages: https://github.com/phetsims/chipper/issues/979
natural-selection 1.1 phet
  deployed: 1.1.1-rc.1
  pushedMessages: https://github.com/phetsims/chipper/issues/979
ph-scale 1.4 phet,phet-io (unreleased)
  pushedMessages: https://github.com/phetsims/chipper/issues/979
ph-scale-basics 1.4 phet,phet-io (unreleased)
  pushedMessages: https://github.com/phetsims/chipper/issues/979
states-of-matter 1.2 phet,phet-io
  deployed: 1.2.1-rc.1
  pushedMessages: https://github.com/phetsims/chipper/issues/979
states-of-matter-basics 1.2 phet,phet-io (unreleased)
  pushedMessages: https://github.com/phetsims/chipper/issues/979
[chipper] https://github.com/phetsims/chipper/issues/979
  d7aea3ba13a73243817dc1254640a8696e284c52
  36aeba695faeccd1b2263fef3a023ddc2db94f64
```

The only sims that `Maintenance.listLinks()` printed were:
  * molecules-and-light 1.5 (https://github.com/phetsims/chipper/issues/979)
  * natural-selection 1.1 (https://github.com/phetsims/chipper/issues/979)
  * states-of-matter 1.2 (https://github.com/phetsims/chipper/issues/979)


So I assume that they will be published when we are ready for `Maintenance.deployProduction()`

These sims don't need more work because they are still in RC:
  * energy-forms-and-changes 1.4 phet,phet-io (unreleased)
  * energy-skate-park 1.1 phet,phet-io (unreleased)
  * ph-scale 1.4 phet,phet-io (unreleased)
  * ph-scale-basics 1.4 phet,phet-io (unreleased)

- [ ] These RC branches have been ditched, and don't need to be maintained (newer rc versions/branches were created), @jonathanolson what should we do for this step?
  * energy-forms-and-changes 1.3 phet,phet-io (unreleased)
  * energy-skate-park 1.0 phet,phet-io (unreleased)


It isn't totally clear to me where the following three sims lie in this process:

@jbphet have these not been published to production yet? I didn't see them on the PhET website. If not I think nothing else is needed for these sims.
* atomic-interactions 1.2 phet,phet-io (unreleased)
* states-of-matter-basics 1.2 phet,phet-io (unreleased)


@jonathanolson  this was published as a "prototype" and can be seen at https://phet.colorado.edu/sims/html/collision-lab/1.0.1/collision-lab_en.html. Why didn't this get into the list to be tested and deployed to production?
* collision-lab 1.0 phet (unreleased)


Over to @jbphet and @jonathanolson to answer the above questions. And also for @jonathanolson to generally review the steps this far. Everything went quite smoothly.

**Comment by zepumph** [comment]
Alright, after https://github.com/phetsims/QA/issues/546 looked good, I went ahead with `Maintenance.deployProduction()`. It worked and published the following sims:

MAL-1.5.5
NS-1.1.1
SOM-1.2.1 (phet-io also)

From here, I still have the above questions for @jonathanolson and @jbphet, which feel like high priority just to finish this issue off and not leave an MR half way complete.

I also want to ping @jbphet and @pixelzoom that now NS should be translatable on the website, and to feel free to test it (as I don't know how to). Unassigning.

**Comment by pixelzoom** [comment]
I don't know how to test translation via the website, so I'm going to unassign myself and request that @jbphet have a look.

**Comment by jonathanolson** [comment]
> These RC branches have been ditched, and don't need to be maintained (newer rc versions/branches were created), @jonathanolson what should we do for this step?

Are those branches we should consider deleting? Or just let them be patched for me.

**Comment by jonathanolson** [comment]
> @jonathanolson this was published as a "prototype" and can be seen at https://phet.colorado.edu/sims/html/collision-lab/1.0.1/collision-lab_en.html. Why didn't this get into the list to be tested and deployed to production?

Prototypes aren't set up at all for maintenance, they're a "publish from dev versions from master" if I'm correct. So the process is purely a manual "figure out a new dev release, have QA test it, etc."

**Comment by zepumph** [comment]
>Prototypes aren't set up at all for maintenance, they're a "publish from dev versions from master" if I'm correct. 

Perhaps the first few were like that, but I believe that NS 1.1 was considered a Prototype, and that is on branches and considered a production release, so maybe this is just something to answer for collision lab. @ariel-phet, do we want this patch to allow translation in the Collision Lab prototype?

**Comment by jonathanolson** [comment]
Maintenance relies on what the website says are production simulations. https://phet.colorado.edu/services/metadata/1.0/simulations?format=json&summary&type=html&locale=en says natural-selection IS a production sim, and that collision-lab is NOT a production sim.

**Comment by pixelzoom** [comment]
To clarify... NS is a production sim that was put through a prototype QA process to save co$t.  It has a visible sim page on the PhET website.  A prototype does not have a visible sim page.

**Comment by jbphet** [comment]
I just examined the HTML for the published versions of natural-selection, molecules-and-light, and states-of-matter, and the problem appears to be corrected, i.e. the license info, strings, etc are no longer minified.  I can also see there there have been several successful submissions of translations for natural-selection.  Finally, I checked the commit history in the babel repo and can see that there was also a successful submission of a translation for molecules-and-light.  These tests have me convinced that the problem is resolved.  Thanks @zepumph and @jonathanolson.

As for the questions that are directed to me in [the comment above](https://github.com/phetsims/chipper/issues/979#issuecomment-686899115), yes, atomic-interactions 1.2 and states-of-matter-basics 1.2 should be patched.  These are not yet published, but will be, probably within the next couple of months.

I think that's it for my part, so I'm unassigning myself.
